### PR TITLE
Change the label checker action

### LINKED ISSUE #344: Replace unmaintained `sugarshin/required-labels-action` GH action
## Problem
We're getting warnings from workflows using this action, see for example https://github.com/WordPress/openverse-catalog/actions/runs/3653606401

No further support seems to be forthcoming since the [last update was on February 2020](https://github.com/sugarshin/required-labels-action/releases). So we would need to replace it sooner or later.

## Description

Change to use [agilepathway/label-checker](https://github.com/agilepathway/label-checker) or any other updated action.

### PR DESCRIPTION
**Pull request by sharmadhiraj86**
## Fixes

Fixes #344 by @krysal 
 
## Description

Modified the label action to agile pathway.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .github/workflows/pr_label_check.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@ name: PR label check
 on:
   pull_request:
     types:
+      - opened
+      - synchronize
+      - reopened
```
Comment (dhruvkb) [inline .github/workflows/pr_label_check.yml:9]: There is no need to add these events.

File: .github/workflows/pr_label_check.yml [hunk 2]
```diff
@@ -13,24 +16,24 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Check aspect label
-        uses: sugarshin/required-labels-action@v0.3.2
+        uses: agilepathway/label-checker@v1.2.9
         with:
-          required_oneof: "💻 aspect: code,📄 aspect: text,🤖 aspect: dx,🕹 aspect: interface,♿️ aspect: a11y"
+          any_of: api,bug,build,dependencies,documentation,enhancement,no-changelog,refactoring
```
Comment (dhruvkb) [inline .github/workflows/pr_label_check.yml:21]: `any_of` is correct but please keep the old list of labels.

**Review by dhruvkb — CHANGES_REQUESTED** [review]

File: .github/workflows/pr_label_check.yml (hunk 1 above)
Comment (sharmadhiraj86) [reply .github/workflows/pr_label_check.yml]: These three have been removed from the list of options.

File: .github/workflows/pr_label_check.yml (hunk 2 above)
Comment (sharmadhiraj86) [reply .github/workflows/pr_label_check.yml]: The new labels have been removed and old labels have been restored.

File: .github/workflows/pr_label_check.yml [hunk 3]
```diff
@@ -13,24 +13,24 @@ jobs:
...
   check_goal_label:
     name: Check goal label
     runs-on: ubuntu-latest
     steps:
       - name: Check goal label
-        uses: sugarshin/required-labels-action@v0.3.2
+        uses: agilepathway/label-checker@v1.2.9
         with:
-          required_oneof: "🌟 goal: addition,🛠 goal: fix,✨ goal: improvement,🧰 goal: internal improvement"
+          any_of: "💻 aspect: code,📄 aspect: text,🤖 aspect: dx,🕹 aspect: interface,♿️ aspect: a11y"
```
Comment (dhruvkb) [inline .github/workflows/pr_label_check.yml:27]: These are supposed to be "goal:*" labels.

File: .github/workflows/pr_label_check.yml [hunk 4]
```diff
@@ -13,24 +13,24 @@ jobs:
...
   check_priority_label:
     name: Check priority label
     runs-on: ubuntu-latest
     steps:
       - name: Check priority label
-        uses: sugarshin/required-labels-action@v0.3.2
+        uses: agilepathway/label-checker@v1.2.9
         with:
-          required_oneof: "🟩 priority: low,🟨 priority: medium,🟧 priority: high,🟥 priority: critical"
+          any_of: "💻 aspect: code,📄 aspect: text,🤖 aspect: dx,🕹 aspect: interface,♿️ aspect: a11y"
```
Comment (dhruvkb) [inline .github/workflows/pr_label_check.yml:36]: These are supposed to be "aspect:*" labels.

**Review by dhruvkb — CHANGES_REQUESTED** [review]

File: .github/workflows/pr_label_check.yml (hunk 4 above)
Comment (sharmadhiraj86) [reply .github/workflows/pr_label_check.yml:36]: Shouldnt this be priority *.

**Comment by sharmadhiraj86** [comment]
The required changes have been updated please check and let me know if there is any further changes required.

**Review by dhruvkb — APPROVED** [review]
Looks good, thanks @sharmadhiraj86!

File: .github/workflows/pr_label_check.yml [hunk 5]
```diff
@@ -13,24 +13,24 @@ jobs:
...
   check_priority_label:
     name: Check priority label
     runs-on: ubuntu-latest
     steps:
       - name: Check priority label
-        uses: sugarshin/required-labels-action@v0.3.2
+        uses: agilepathway/label-checker@v1.2.9
         with:
-          required_oneof: "🟩 priority: low,🟨 priority: medium,🟧 priority: high,🟥 priority: critical"
+          any_of: "🟩 priority: low,🟨 priority: medium,🟧 priority: high,🟥 priority: critical"
```
Comment (sarayourfriend) [inline .github/workflows/pr_label_check.yml:36]:
```suggestion
          one_of: "🟩 priority: low,🟨 priority: medium,🟧 priority: high,🟥 priority: critical"
```

**Review by sarayourfriend — APPROVED** [review]
This LGTM but I will avoid merging it yet to ask @dhruvkb whether these should all be `any_of` or if some of them (priority, I think, at least?) should be `one_of`?

I left a suggested change for it in case that is the intention. `any_of` appears to be >=1 which might be okay for aspect and goal, but I don't think it makes sense for priority :thinking:

**Comment by sharmadhiraj86** [comment]
@dhruvkb Please check whether the suggested change should be done or not. Thanks!

File: .github/workflows/pr_label_check.yml (hunk 5 above)
Comment (dhruvkb) [reply .github/workflows/pr_label_check.yml:36]: I agree with you @sarayourfriend, good catch!

**Comment by sharmadhiraj86** [comment]
@sarayourfriend Please merge the PR.

**Comment by krysal** [comment]
Thanks for the help, @sharmadhiraj86!

**Pull request merged**